### PR TITLE
Stac 11357 snow filtering with query

### PR DESCRIPTION
**Pull request by hruhek**
### Step 1: Link to Jira issue
https://stackstate.atlassian.net/browse/STAC-11357

### Step 2: Description of changes
Adding filtering with sysparm_query parameter

### Step 3: Did you add / update tests for your changes in the right area?
- [ ] Unit/Component test

### Step 4: I'm confident that everything is properly tested:
I got a PO / QA Approval by:
- [ ] Johnny

### Step 5: Can we ship this feature to production?
- [ ] Yes, I'm proud of my work. Ship it! :ship:

### CODE REVIEW

File: servicenow/stackstate_checks/servicenow/data/conf.yaml.example [hunk 1]
```diff
@@ -22,6 +22,18 @@ instances:
     # Verify https. Default value is true.
     # verify_https: true
 
+    # ServiceNow Custom filter. Check ServiceNow documentation about filtering with sysparm_query parameter.
+    # https://developer.servicenow.com/dev.do#!/learn/learning-plans/orlando/servicenow_application_developer/app_store_learnv2_rest_orlando_more_about_query_parameters
+
+    # ServiceNow CMDB Configuration Items query. Default value is undefined.
```
Comment (TizianoPerrucci) [inline servicenow/stackstate_checks/servicenow/data/conf.yaml.example:28]: `Default value is undefined` means if `None` no filter will be applied?
can we make the comment a bit more clear?

Also this applies to topology component listing, right? I would also add that in the comment.

File: servicenow/stackstate_checks/servicenow/data/conf.yaml.example [hunk 2]
```diff
@@ -22,6 +22,18 @@ instances:
...
+    # cmdb_ci_sysparm_query: company.nameSTARTSWITHaxa
+
+    # ServiceNow CMDB Configuration Items Relations query. Default value is undefined.
```
Comment (TizianoPerrucci) [inline servicenow/stackstate_checks/servicenow/data/conf.yaml.example:31]: `Default value is undefined` means if `None` no filter will be applied?
can we make the comment a bit more clear?

Also this applies to topology relation listing, right? I would also add that in the comment.

File: servicenow/stackstate_checks/servicenow/data/conf.yaml.example [hunk 3]
```diff
@@ -22,6 +22,18 @@ instances:
...
+    # cmdb_rel_ci_sysparm_query: parent.company.nameSTARTSWITHaxa^ORchild.company.nameSTARTSWITHaxa
+
+    # ServiceNow Change Request query. Default value is undefined.
```
Comment (TizianoPerrucci) [inline servicenow/stackstate_checks/servicenow/data/conf.yaml.example:34]: `Default value is undefined` means if `None` no filter will be applied?
can we make the comment a bit more clear?

Also this applies to topology event listing, right? I would also add that in the comment.

File: servicenow/stackstate_checks/servicenow/servicenow.py [hunk 4]
```diff
@@ -170,14 +175,11 @@ def _batch_collect_components(self, instance_info, offset):
...
         auth = (instance_info.user, instance_info.password)
         url = instance_info.url + '/api/now/table/cmdb_ci'
-        sys_class_filter_query = self.get_sys_class_component_filter_query(instance_info.include_resource_types)
-        if sys_class_filter_query:
-            params = {'sysparm_query': sys_class_filter_query}
-        else:
-            params = {}
+        sys_class_filter_query = self._get_sys_class_component_filter_query(instance_info.include_resource_types)
+        params = self._params_append_to_sysparm_query(add_to_query=sys_class_filter_query)
+        params = self._params_append_to_sysparm_query(add_to_query=instance_info.cmdb_ci_sysparm_query, params=params)
```
Comment (TizianoPerrucci) [inline servicenow/stackstate_checks/servicenow/servicenow.py:182]: I miss a test when batch collecting components which ensure that the `sysparm` is set correctly if we configure the query filtering

File: servicenow/stackstate_checks/servicenow/servicenow.py [hunk 5]
```diff
@@ -267,16 +269,11 @@ def _batch_collect_relations(self, instance_info, offset):
...
-        if sys_class_filter_query:
-            params = {
-                'sysparm_query': sys_class_filter_query
-            }
-        else:
-            params = {}
-
+        sys_class_filter_query = self._get_sys_class_relation_filter_query(instance_info.include_resource_types)
+        params = self._params_append_to_sysparm_query(add_to_query=sys_class_filter_query)
+        params = self._params_append_to_sysparm_query(add_to_query=instance_info.cmdb_rel_ci_sysparm_query,
```
Comment (TizianoPerrucci) [inline servicenow/stackstate_checks/servicenow/servicenow.py:274]: I miss a test when collecting batch relations which ensure that the `sysparm` is set correctly if we configure the query filtering

File: servicenow/stackstate_checks/servicenow/servicenow.py [hunk 6]
```diff
@@ -308,6 +305,7 @@ def _collect_change_requests(self, instance_info):
             'sysparm_limit': instance_info.change_request_process_limit,
             'sysparm_query': sysparm_query
         }
+        params = self._params_append_to_sysparm_query(instance_info.change_request_sysparm_query, params)
```
Comment (TizianoPerrucci) [inline servicenow/stackstate_checks/servicenow/servicenow.py:308]: I miss a test when collecting topology events which ensure that the `sysparm` is set correctly if we configure the query filtering

**Review by TizianoPerrucci — CHANGES_REQUESTED** [review]

File: servicenow/stackstate_checks/servicenow/servicenow.py (hunk 6 above)
Comment (vishalxebia) [reply servicenow/stackstate_checks/servicenow/servicenow.py:308]: added.

File: servicenow/stackstate_checks/servicenow/servicenow.py (hunk 5 above)
Comment (vishalxebia) [reply servicenow/stackstate_checks/servicenow/servicenow.py:274]: added.

File: servicenow/stackstate_checks/servicenow/servicenow.py (hunk 4 above)
Comment (vishalxebia) [reply servicenow/stackstate_checks/servicenow/servicenow.py:182]: added.

File: servicenow/stackstate_checks/servicenow/data/conf.yaml.example (hunk 3 above)
Comment (vishalxebia) [reply servicenow/stackstate_checks/servicenow/data/conf.yaml.example]: Added that comment.

File: servicenow/stackstate_checks/servicenow/data/conf.yaml.example (hunk 2 above)
Comment (vishalxebia) [reply servicenow/stackstate_checks/servicenow/data/conf.yaml.example]: Added the same.

File: servicenow/stackstate_checks/servicenow/data/conf.yaml.example (hunk 1 above)
Comment (vishalxebia) [reply servicenow/stackstate_checks/servicenow/data/conf.yaml.example]: Added the same.

**Review by vishalxebia — APPROVED** [review]

**Pull request merged**